README.md

```diff
@@ -1,51 +1,100 @@
 
-Deployed https://explorer.testnet.near.org/transactions/HpyWnTM4KUnt45MoQiAgwco2pMSQqDmoSn2CwtU6N4mU
+### Prerequisites
 
-Some votes from this contract:
+[Install Rust](https://www.rust-lang.org/tools/install)
 
-First vote:
+```
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
 
-https://explorer.testnet.near.org/transactions/CRaR8mLaUpyRqrXWsq4jY1NTjGTviZta8xAKp4f3r2W1
+Install `near-cli` (NEAR command line)
 
-Trying to vote with the same account in the same proposal you can´t:
+```
+npm install -g near-cli
+```
 
-https://explorer.testnet.near.org/transactions/C8XkEP93VHiTWTzbtRXUYMxFfXNQ58TqRysPZyJXBjRM
+### Create NEAR Wallet
 
-Voting from a different account:
+You need to do this on the testnet otherwise you'll be paying for gas fees.
 
-https://explorer.testnet.near.org/transactions/6rUuM8LHJUBWA9FYhDncF7Z3hkijb6UPLvTcgov7WcDL
+[wallet.testnet.near.org](https://wallet.testnet.near.org/)
 
+Write your wallet address somewhere, we'll need it later. It should look something like this: `mytestaccount.testnet`
 
-To vote you need to have the near-cli installed and be logged in:
 
-To vote for NFT MAKER APP:
 
-```near call votingnico.testnet vote '{"proposal_name":"NFT MAKER APP"}' --accountId YOUR_ACCOUNT.testnet```
 
-To vote for Homepage: 
 
-```near call votingnico.testnet vote '{"proposal_name":"HomePage"}' --accountId YOUR_ACCOUNT.testnet```
+### Building the contract
 
+First download the code from this repo.
 
-To test this contract:
+```
+git clone git@github.com:fioritonicolas/voting-contract.git
+cd voting-contract
+```
 
-```cargo test -- --nocapture```
+Next test and compile the rust code
 
-To build this contract:
+```
+cargo test -- --nocapture
+cargo build --target wasm32-unknown-unknown --release
 
-```cargo build --target wasm32-unknown-unknown --release```
+```
 
-To deploy this contract:
+Excellent, this should give you something to deploy. Lets deploy the contract to the testnet. This is where you will need your wallet address from before.
 
-```near deploy --wasmFile target/wasm32-unknown-unknown/release/voting_contract.wasm --accountId votingnico.testnet```
+```
+near deploy --wasmFile target/wasm32-unknown-unknown/release/voting_contract.wasm --accountId mytestaccount.testnet
+```
+
+This should return with a URL that lets you view your contract on the blockchain. It will be a transaction like this:
+
+[explorer.testnet.near.org/transactions/HpyWnTM4KUnt45MoQiAgwco2pMSQqDmoSn2CwtU6N4mU](https://explorer.testnet.near.org/transactions/HpyWnTM4KUnt45MoQiAgwco2pMSQqDmoSn2CwtU6N4mU)
+
+🚨 It's important to note that your wallet address and contract name are the same! 
+
+### Voting on the contract
+
+First you need to login
+
+```
+near-cli login
+```
+
+Right now there are two proposals that you can vote on:
+
+1. NFT MAKER APP
+2. HomePage
+
+Here's how you vote for NFT Maker
+
+```
+near call mytestaccount.testnet vote '{"proposal_name":"NFT MAKER APP"}' --accountId mytestaccount.testnet
+```
+
```

Comment:
what about if we explain these params. It took me a while to understand it, and is not that clear on the near docs either.

Something like:

near call receives first the contract name, then one of the public functions in the contract, if this function accepts parameters next to the function name should be passed, and last the signing account which is calling this contract, this account is has to be the same that you logged in in the previous step.